avocado/core/test.py

```diff
@@ -1029,6 +1029,33 @@ def fetch_asset(self, name, asset_hash=None, algorithm=None,
             # otherwise re-throw OSError
             raise e
 
+    def get_asset_by_name(self, name, expire=None, asset_hash=None,
```

Comment:
You're right that this is the most controversial change... and I'm not convinced that it's needed *or* that it's the commit that fixes #4313.  IIUC, without `get_asset_by_name()` the same feature set would still be available for users of `fetch_asset()`, or am I missing something?  Maybe I am, so quick questions:

What is the behavior of `fetch_asset()` being given only the name?

What is the downside of the plugin trying to fetch the asset by name?

Also, adding an `avocado.core.test.Test` API should be done with extreme caution, as the one area we try our best to guarantee stability.

Reply:
Hi, @clebergnu thanks for waiting for this reply. I got preempted with other tasks.

> You're right that this is the most controversial change... and I'm not convinced that it's needed _or_ that it's the commit that fixes #4313. IIUC, without `get_asset_by_name()` the same feature set would still be available for users of `fetch_asset()`, or am I missing something? Maybe I am, so quick questions:
> 
> What is the behavior of `fetch_asset()` being given only the name?

AFAICT,  `plugins/assets.py` is using `"fetch_asset"` string as `PATTERN` and will try to download anyway, since `name` (on the current logic) could also be an 'url', according to the `fetch_asset()` docstring. Now, assuming that name is not a URL, and we call it only with an already registered generic name, `utils/asset.py` will try to "fetch it" ("download") and it will fail.

Because of that, previous commits are not delivering the same feature to our users as this commit it is.

> What is the downside of the plugin trying to fetch the asset by name?

We would be using the method as it was not designed to be executed. Unnecessary error messages could be displayed to the users/logs, giving them the impression that something is trying to be downloaded. On top of that, we could be fixing a design by adding patches and too many implicit assumptions (more than what already have today). IMO this is a bad design choice. Like I said before this entire asset logic, as it is, wasn't designed thinking about today's user cases, and making patches on top of it could lead us to a never-ending fix with a hard to scale code.

> Also, adding an `avocado.core.test.Test` API should be done with extreme caution, as the one area we try our best to guarantee stability.

I completely understand and agree with you here. However, in order to grow and adapt to changes, IMO we need to be flexible here (as long we take the right deprecations measures when necessary). As I can see, we are being careful with our users, we aren't removing features or changing existing calls, just adding a new method to our API based on a new feature request, and that is normal for a healthy API.